Linux pre-compiled builds are dynamically linked to the Nix store

musl build required — onig giving us a little bit of trouble:

```
error: build of '/nix/store/y82xac3dxrws2rzis3yldir2a1pcgjzk-comrak-0.47.0.drv' on 'ssh-ng://rosetta-builder' failed: builder for '/nix/store/y82xac3dxrws2rzis3yldir2a1pcgjzk-comrak-0.47.0.drv' failed with exit code 101;
       last 25 log lines:
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: regerror.c:(.text.onig_error_code_to_str+0x360): undefined reference to `__snprintf_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: regerror.c:(.text.onig_error_code_to_str+0x390): undefined reference to `__snprintf_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: /build/rustc5rK0WZ/libonig_sys-e001fc07d29eb553.rlib(regerror.o):regerror.c:(.text.onig_error_code_to_str+0x3c0): more undefined references to `__snprintf_chk' follow
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: /build/rustc5rK0WZ/libonig_sys-e001fc07d29eb553.rlib(regerror.o): in function `onig_error_code_to_str':
       >           regerror.c:(.text.onig_error_code_to_str+0x418): undefined reference to `__memcpy_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: /build/rustc5rK0WZ/libonig_sys-e001fc07d29eb553.rlib(regerror.o): in function `onig_snprintf_with_pattern':
       >           regerror.c:(.text.onig_snprintf_with_pattern+0x9c): undefined reference to `__vsnprintf_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: regerror.c:(.text.onig_snprintf_with_pattern+0x1dc): undefined reference to `__snprintf_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: regerror.c:(.text.onig_snprintf_with_pattern+0x558): undefined reference to `__snprintf_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: /build/rustc5rK0WZ/libonig_sys-e001fc07d29eb553.rlib(regparse.o): in function `prs_callout_args.constprop.0.isra.0':
       >           regparse.c:(.text.prs_callout_args.constprop.0.isra.0+0x108): undefined reference to `__memcpy_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: /build/rustc5rK0WZ/libonig_sys-e001fc07d29eb553.rlib(regcomp.o): in function `compile_tree':
       >           regcomp.c:(.text.compile_tree+0x330): undefined reference to `__memcpy_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: regcomp.c:(.text.compile_tree+0x990): undefined reference to `__memcpy_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: /build/rustc5rK0WZ/libonig_sys-e001fc07d29eb553.rlib(regcomp.o): in function `onig_compile':
       >           regcomp.c:(.text.onig_compile+0x438): undefined reference to `__memcpy_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: regcomp.c:(.text.onig_compile+0x948): undefined reference to `__memcpy_chk'
       >           /nix/store/ll82pfjsgww6cmi7mazcpsbv4b1szd9b-binutils-2.44/bin/ld: /build/rustc5rK0WZ/libonig_sys-e001fc07d29eb553.rlib(regenc.o):regenc.c:(.text.onigenc_strdup+0x44): more undefined references to `__memcpy_chk' follow
       >           collect2: error: ld returned 1 exit status
       >
       >   = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
       >   = note: use the `-l` flag to specify native libraries to link
       >   = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-link-lib)
       >
       > error: could not compile `comrak` (bin "comrak") due to 1 previous error
```

--- a/flake.nix
+++ b/flake.nix
@@ -3,7 +3,6 @@
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-25.05";
-    crane.url = "github:ipetkov/crane";
     fenix = {
       url = "github:nix-community/fenix";
       inputs.nixpkgs.follows = "nixpkgs";
@@ -12,10 +11,9 @@
 
   outputs =
     {
-      self,
       nixpkgs,
-      crane,
       fenix,
+      ...
     }:
     let
       systems = [
@@ -25,88 +23,34 @@
         "x86_64-linux"
       ];
       eachSystem = nixpkgs.lib.genAttrs systems;
-
-      mkComrak =
-        system:
-        let
-          pkgs = nixpkgs.legacyPackages.${system};
-          craneLib = crane.mkLib pkgs;
-          src = craneLib.cleanCargoSource (craneLib.path ./.);
-
-          commonArgs = {
-            inherit src;
-
-            buildInputs = nixpkgs.lib.optionals pkgs.stdenv.isDarwin [ pkgs.libiconv ];
-          };
-
-          cargoArtifacts = craneLib.buildDepsOnly commonArgs;
-
-          comrak = craneLib.buildPackage (
-            commonArgs
-            // {
-              inherit cargoArtifacts;
-
-              doCheck = false;
-            }
-          );
-        in
-        {
-          inherit
-            craneLib
-            src
-            commonArgs
-            cargoArtifacts
-            comrak
-            ;
-        };
-
     in
     {
 
-      checks = eachSystem (
+      packages = eachSystem (
         system:
         let
-          inherit (mkComrak system)
-            craneLib
-            src
-            commonArgs
-            cargoArtifacts
-            comrak
-            ;
+          pkgs = nixpkgs.legacyPackages.${system};
+          cargoToml = pkgs.lib.importTOML ./Cargo.toml;
+
+          mkComrak =
+            pkgs:
+            pkgs.rustPlatform.buildRustPackage {
+              pname = "comrak";
+              inherit (cargoToml.package) version;
+
+              src = ./.;
+              cargoLock.lockFile = ./Cargo.lock;
+
+              doCheck = false;
+            };
         in
-        {
-          inherit comrak;
+        rec {
+          default = comrak;
 
-          comrak-clippy = craneLib.cargoClippy (
-            commonArgs
-            // {
-              inherit cargoArtifacts;
-              # cargoClippyExtraArgs = "--lib --bins --examples --tests -- --deny warnings";
-              # XXX Not sure if we can fix all these and retain our current MSRV.
-              cargoClippyExtraArgs = "--lib --bins --examples --tests";
-            }
-          );
-
-          comrak-doc = craneLib.cargoDoc (commonArgs // { inherit cargoArtifacts; });
-
-          comrak-fmt = craneLib.cargoFmt { inherit src; };
-
-          comrak-nextest = craneLib.cargoNextest (
-            commonArgs
-            // {
-              inherit cargoArtifacts;
-              partitions = 1;
-              partitionType = "count";
-            }
-          );
+          comrak = mkComrak pkgs;
+          comrak-musl = mkComrak pkgs.pkgsStatic;
         }
       );
-
-      packages = eachSystem (system: rec {
-        default = comrak;
-
-        inherit (mkComrak system) comrak;
-      });
 
       formatter = eachSystem (system: nixpkgs.legacyPackages.${system}.nixfmt-rfc-style);
 

--- a/script/build-releases
+++ b/script/build-releases
@@ -10,14 +10,24 @@ function build {
     local rust_ver="$1"
     local flake_ref="$2"
 
-    echo "--- building $flake_ref"
-    nix build ".#$flake_ref" &&
-        cp result/bin/comrak "comrak-$version-$rust_ver"
+    local target="comrak-$version-$rust_ver"
+    local nix="nix"
+    if command -v nom &>/dev/null; then
+        nix="nom"
+    fi
+
+    if ! test -f "$target"; then
+        echo "--- building $flake_ref"
+        "$nix" build ".#$flake_ref" &&
+            cp result/bin/comrak "$target"
+    else
+        echo "--- skipping $flake_ref, $target already exists"
+    fi
 }
 
 build aarch64-apple-darwin packages.aarch64-darwin.default
 build x86_64-apple-darwin packages.x86_64-darwin.default
-build aarch64-unknown-linux-gnu packages.aarch64-linux.default
-build x86_64-unknown-linux-gnu packages.x86_64-linux.default
+build aarch64-unknown-linux-musl packages.aarch64-linux.comrak-musl
+build x86_64-unknown-linux-musl packages.x86_64-linux.comrak-musl
 
 file "comrak-$version-"*

